src/state/transactions/updater.tsx

```diff
@@ -71,13 +71,79 @@ export default function Updater(): null {
     [chainId, library]
   )
 
+  /**
+   * Returns order from the Gnosis Protocol API
+   */
+  const getGnosisProtocolOrder = useCallback(
+    (orderId: string) => {
+      if (!chainId) throw new Error('No library or chainId')
+      const retryOptions = RETRY_OPTIONS_BY_CHAIN_ID[chainId] ?? DEFAULT_RETRY_OPTIONS
+      return retry(
+        () =>
+          fetch(`${GnosisProtocolTrade.getApi(chainId).baseUrl}/api/v1/orders/${orderId}`).then(res => {
+            if (!res.ok) {
+              console.debug('Retrying for order ', orderId)
+              throw new RetryableError()
+            }
+
+            return res.json()
+          }),
+        retryOptions
+      )
```

Comment:
In my opinion if we use `async/await` in the code we should stick to it. This code is easier to read. I understand that few lines above is similar code using `.then` but it also should be refactored to `async/await`

```suggestion
    return retry(async () => {
        const res = await fetch(`${GnosisProtocolTrade.getApi(chainId).baseUrl}/api/v1/orders/${orderId}`)
        if (!res.ok) {
          console.debug('Retrying for order ', orderId)
          throw new RetryableError()
        }
        return res.json()
      }, retryOptions)
```